packages/grid/x-data-grid-pro/src/tests/columnSpanning.DataGridPro.test.tsx

```diff
@@ -0,0 +1,261 @@
+  fireEvent.mouseUp(cell);
+  fireEvent.click(cell);
+}
+
+describe('<DataGridPro /> - Column Spanning', () => {
+  const { render } = createRenderer({ clock: 'fake' });
+
+  const baselineProps = {
+    rows: [
+      {
+        id: 0,
+        brand: 'Nike',
+        category: 'Shoes',
+        price: '$120',
+        rating: '4.5',
+      },
+      {
+        id: 1,
+        brand: 'Adidas',
+        category: 'Shoes',
+        price: '$100',
+        rating: '4.5',
+      },
+      {
+        id: 2,
+        brand: 'Puma',
+        category: 'Shoes',
+        price: '$90',
+        rating: '4.5',
+      },
+    ],
+  };
+
+  it('should not apply `colSpan` in pinned columns section if there is only one column there', function test() {
+    if (isJSDOM) {
+      // Need layouting
+      this.skip();
+    }
+
+    render(
+      <div style={{ width: 500, height: 300 }}>
+        <DataGridPro
+          {...baselineProps}
+          columns={[
+            { field: 'brand', colSpan: 2, width: 110 },
+            { field: 'category' },
+            { field: 'price' },
+          ]}
+          initialState={{ pinnedColumns: { left: ['brand'], right: [] } }}
+        />
+      </div>,
+    );
+
+    expect(getCell(0, 0).offsetWidth).to.equal(110);
+    expect(() => getCell(0, 0)).to.not.throw();
+    expect(() => getCell(0, 1)).to.not.throw();
+    expect(() => getCell(0, 2)).to.not.throw();
+  });
+
+  it('should apply `colSpan` inside pinned columns section', () => {
+    render(
+      <div style={{ width: 500, height: 300 }}>
+        <DataGridPro
+          {...baselineProps}
+          columns={[{ field: 'brand', colSpan: 2 }, { field: 'category' }, { field: 'price' }]}
+          initialState={{ pinnedColumns: { left: ['brand', 'category'], right: [] } }}
+        />
+      </div>,
+    );
+
+    expect(() => getCell(0, 0)).to.not.throw();
+    expect(() => getCell(0, 1)).to.throw(/not found/);
+    expect(() => getCell(0, 2)).to.not.throw();
+  });
+
+  /* eslint-disable material-ui/disallow-active-element-as-key-event-target */
+  describe('key navigation', () => {
+    const columns: GridColDef[] = [
+      { field: 'brand', colSpan: ({ row }) => (row.brand === 'Nike' ? 2 : 1) },
+      { field: 'category', colSpan: ({ row }) => (row.brand === 'Adidas' ? 2 : 1) },
+      { field: 'price', colSpan: ({ row }) => (row.brand === 'Puma' ? 2 : 1) },
+      { field: 'rating' },
+    ];
+
+    it('should work after column reordering', () => {
+      let apiRef: React.MutableRefObject<GridApi>;
+
+      const Test = () => {
+        apiRef = useGridApiRef();
+
+        return (
+          <div style={{ width: 500, height: 300 }}>
+            <DataGridPro apiRef={apiRef} {...baselineProps} columns={columns} />
+          </div>
+        );
+      };
+
+      render(<Test />);
+
+      apiRef!.current.setColumnIndex('price', 1);
+
+      fireClickEvent(getCell(1, 1));
+      fireEvent.keyDown(document.activeElement!, { key: 'ArrowRight' });
```

Comment:
This allows to remove `eslint-disable` above.

```suggestion
      fireEvent.keyDown(getCell(1, 1), { key: 'ArrowRight' });
```

Other `document.activeElement` could also be removed.

Reply:
Personally, I prefer using `document.activeElement`, since it allows to make sure that the right element is focused.

Consider this case:
```js
// start editing
fireEvent.keyDown(document.activeElement!, { key: 'Enter' });

fireEvent.keyDown(document.activeElement!, { key: 'Tab', shiftKey: true });
await waitFor(() => {
  expect(getActiveCell()).to.equal('0-0');
});
```

Replacing `document.activeElement` with cell element and input element would work, but if for some reason input element won't get focus after pressing `Enter` - the test won't fail.

So I would rather keep `document.activeElement`.
What do you think?

Reply:
This is a ongoing discussion here. I prefer to not disable the ESLint rules to be consistent with how the tests are written in the core repo. I try to avoid `document.activeElement` when writing new tests.

> Personally, I prefer using document.activeElement, since it allows to make sure that the right element is focused.

Note that we already have a check to ensure that the element passed is the one with focus: https://github.com/mui/material-ui/blob/0251fd5222079e4a512bbd32dff4a7a4c3efe997/test/utils/createRenderer.tsx#L628

`document.activeElement` bypasses this check.

> Replacing document.activeElement with cell element and input element would work, but if for some reason input element won't get focus after pressing Enter - the test won't fail.

I didn't understand the problem. I replaced with the cell element and the input was focused. These tests use the legacy editing API where Shift + Tab may not be well supported. Try with the new API.